pandas/tseries/plotting.py

```diff
@@ -1,344 +1,3 @@
-        if isinstance(freq, DateOffset):
-            freq = freq.rule_code
-
-        if freq is None:
-            freq = _get_ax_freq(ax)
-
-        if freq is None:
-            raise ValueError('Could not get frequency alias for plotting')
-
-        freq = frequencies.get_base_alias(freq)
-        freq = frequencies.get_period_alias(freq)
-
-        data = data.to_period(freq=freq)
-    return data
-
-
-# Patch methods for subplot. Only format_dateaxis is currently used.
-# Do we need the rest for convenience?
-
-def format_timedelta_ticks(x, pos, n_decimals):
-    """
-    Convert seconds to 'D days HH:MM:SS.F'
-    """
-    s, ns = divmod(x, 1e9)
-    m, s = divmod(s, 60)
-    h, m = divmod(m, 60)
-    d, h = divmod(h, 24)
-    decimals = int(ns * 10**(n_decimals - 9))
-    s = r'{:02d}:{:02d}:{:02d}'.format(int(h), int(m), int(s))
-    if n_decimals > 0:
-        s += '.{{:0{:0d}d}}'.format(n_decimals).format(decimals)
-    if d != 0:
-        s = '{:d} days '.format(int(d)) + s
-    return s
-
-
-def format_dateaxis(subplot, freq, index):
-    """
-    Pretty-formats the date axis (x-axis).
-
-    Major and minor ticks are automatically set for the frequency of the
-    current underlying series.  As the dynamic mode is activated by
-    default, changing the limits of the x axis will intelligently change
-    the positions of the ticks.
-    """
-
-    # handle index specific formatting
-    # Note: DatetimeIndex does not use this
-    # interface. DatetimeIndex uses matplotlib.date directly
-    if isinstance(index, PeriodIndex):
-
-        majlocator = TimeSeries_DateLocator(freq, dynamic_mode=True,
-                                            minor_locator=False,
-                                            plot_obj=subplot)
-        minlocator = TimeSeries_DateLocator(freq, dynamic_mode=True,
-                                            minor_locator=True,
-                                            plot_obj=subplot)
-        subplot.xaxis.set_major_locator(majlocator)
-        subplot.xaxis.set_minor_locator(minlocator)
-
-        majformatter = TimeSeries_DateFormatter(freq, dynamic_mode=True,
-                                                minor_locator=False,
-                                                plot_obj=subplot)
-        minformatter = TimeSeries_DateFormatter(freq, dynamic_mode=True,
-                                                minor_locator=True,
-                                                plot_obj=subplot)
-        subplot.xaxis.set_major_formatter(majformatter)
-        subplot.xaxis.set_minor_formatter(minformatter)
-
-        # x and y coord info
-        subplot.format_coord = lambda t, y: (
-            "t = {0}  y = {1:8f}".format(Period(ordinal=int(t), freq=freq), y))
-
-    elif isinstance(index, TimedeltaIndex):
-        subplot.xaxis.set_major_formatter(
-            TimeSeries_TimedeltaFormatter())
-    else:
-        raise TypeError('index type not supported')
-
```

Comment:
Not fully sure, I suppose the idea was back compat? (@sinhrks ?) 

But this one only imports `tsplot`, I don't think we should regard that as public, so can just as well remove this file then.